pkg/controller/serviceaccount/tokens_controller.go

```diff
@@ -193,23 +225,50 @@ func (e *TokensController) queueServiceAccountUpdateSync(oldObj interface{}, new
 	}
 }
 
-func (e *TokensController) requeue(queue *workqueue.Type, key interface{}) {
-	// Prevent hot-looping
-	if queue.Len() < e.activeWorkers {
-		time.Sleep(requeuePeriod)
+func (e *TokensController) retry(queue *workqueue.Type, key interface{}) {
+	// get retry count for this key
+	retryCount := 0
+	if c, ok := e.retryCounts.Get(key); ok {
+		retryCount = c.(int)
+	}
+
+	// stop retrying after N attempts
+	if retryCount >= e.maxRetries {
+		glog.V(4).Infof("retried %d times: %#v", retryCount, key)
+		return
+	}
+
+	// increment our retry count
+	retryCount++
+	e.retryCounts.Add(key, retryCount)
+
+	// See how long we should wait before retrying
+	delay := e.retryRateLimiter.Take(1)
+
+	// Retry immediately
+	if delay == 0 {
+		queue.Add(key)
+		return
 	}
-	queue.Add(key)
+
+	// Or delay
+	glog.V(4).Infof("rate-limiter delay of %s retrying %#v", key, delay)
+	go func() {
+		defer utilruntime.HandleCrash()
+		time.Sleep(delay)
```

Comment:
@deads2k convinced me the fanout wasn't good... I'll rebase once https://github.com/kubernetes/kubernetes/pull/23444 is available and use that instead